protocol/x/feetiers/keeper/keeper.go

```diff
@@ -97,6 +100,15 @@ func (k Keeper) getUserFeeTier(ctx sdk.Context, address string) (uint32, *types.
 		idx = uint32(i)
 	}
 
+	// Bump up to RefereeStartingFeeTier if the user is referred by an affiliate.
+	// We subtract 1 because the fee tiers are 1-indexed.
+	if idx < types.RefereeStartingFeeTier-1 {
```

Comment:
Let's avoid the minus one by making this [change](https://github.com/dydxprotocol/v4-chain/pull/2217/files#r1747252683)